vit-servicing-station-tests/src/tests/graphql/funds.rs

```diff
@@ -0,0 +1,105 @@
+use crate::common::{
+    clients::graphql::templates::{
+        FundByIdWrongArgType, FundWithoutArgument, FundsFieldDoesNotExist, FundsRequiredFields,
+    },
+    startup::quick_start,
+};
+use askama::Template;
+use assert_fs::TempDir;
+
+#[test]
+pub fn get_fund_by_id_test() -> Result<(), Box<dyn std::error::Error>> {
+    let temp_dir = TempDir::new().unwrap();
+    let (server, snapshot) = quick_start(&temp_dir).unwrap();
+
+    let fund_id: i32 = snapshot.funds().first().unwrap().id;
+
+    let graphql_client = server.graphql_client_with_token(&snapshot.token_hash());
+
+    let fund = graphql_client.fund_by_id(fund_id).unwrap();
+    assert_eq!(fund, snapshot.fund_by_id(fund_id).unwrap().clone());
+    Ok(())
+}
+
+#[test]
+pub fn funds_test() -> Result<(), Box<dyn std::error::Error>> {
+    let temp_dir = TempDir::new().unwrap();
+    let (server, snapshot) = quick_start(&temp_dir).unwrap();
+    let graphql_client = server.graphql_client_with_token(&snapshot.token_hash());
+    let mut funds = snapshot.funds();
+    funds.sort_by_key(|k| k.id);
+    assert_eq!(graphql_client.funds().unwrap(), funds);
+    Ok(())
+}
+
+#[test]
+pub fn fund_without_argument_test() -> Result<(), Box<dyn std::error::Error>> {
+    let temp_dir = TempDir::new().unwrap();
+    let (server, snapshot) = quick_start(&temp_dir).unwrap();
+
+    let graphql_client = server.graphql_client_with_token(&snapshot.token_hash());
+    let query_result = graphql_client
+        .run_query(&FundWithoutArgument.render().unwrap())
+        .unwrap();
+    assert!(query_result["errors"][0]["message"]
+        .as_str()
+        .unwrap()
+        .contains("is required but not provided"));
+    Ok(())
+}
+
+#[test]
+pub fn fund_wrong_argument_type_test() -> Result<(), Box<dyn std::error::Error>> {
+    let temp_dir = TempDir::new().unwrap();
+    let (server, snapshot) = quick_start(&temp_dir).unwrap();
+
+    let fund_id: i32 = snapshot.funds().first().unwrap().id;
+    let graphql_client = server.graphql_client_with_token(&snapshot.token_hash());
+    let query_result = graphql_client
+        .run_query(
+            &FundByIdWrongArgType {
+                id: fund_id.to_string(),
+            }
+            .render()
+            .unwrap(),
+        )
+        .unwrap();
+    assert!(query_result["errors"][0]["message"]
+        .as_str()
+        .unwrap()
+        .contains("Invalid value for argument"));
```

Comment:
Can we check the error code too in this tests? I would prefer to rely on the codes than the messages (although we can keep the message checking).
Or are this the `graphql` errors? If they are standard errors derivated from `graphql` instead of our customized ones, then there is no really need of testing them (they should be standard...hopefully).

Reply:
Those are standard (i think) graphql errors. I can remove them.

Reply:
Leave them for now. But just knowing it should fail should be enough. 
If depending on the error messages that are not ours we may have to change them if they change the implementation of whatever lib we are using (or if we need to change lib).